From 9eef0eeeff0eb128fea73e3530e686c0afbdc78a Mon Sep 17 00:00:00 2001
From: Mike Frysinger <vapier@gentoo.org>
Date: Sat, 4 May 2013 14:10:41 -0400
Subject: [PATCH] workaround crash when handling signals in static PIEs

work around ... not entirely sure what is going on here.

2011-03-01	squeezy  <vina@mailserver.eu>

	* sysdeps/unix/sysv/linux/x86_64/sigaction.c fix the __restore_rt symbol

http://bugs.gentoo.org/283470
---
 sysdeps/unix/sysv/linux/x86_64/sigaction.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sysdeps/unix/sysv/linux/x86_64/sigaction.c b/sysdeps/unix/sysv/linux/x86_64/sigaction.c
index efc837f..344ca10 100644
--- a/sysdeps/unix/sysv/linux/x86_64/sigaction.c
+++ b/sysdeps/unix/sysv/linux/x86_64/sigaction.c
@@ -37,7 +37,7 @@
 
 /* Using the hidden attribute here does not change the code but it
    helps to avoid warnings.  */
-extern void restore_rt (void) asm ("__restore_rt") attribute_hidden;
+extern void restore_rt (void) asm ("__restore_rt") __attribute__((__visibility__("hidden")));
 
 
 /* If ACT is not NULL, change the action for SIG to *ACT.
-- 
2.4.4

